tidy3d/components/medium.py

```diff
@@ -471,6 +471,30 @@ class AbstractMedium(ABC, Tidy3dBaseModel):
         description="Modulation spec applied on top of the base medium properties.",
     )
 
+    @classmethod
+    def from_general_nk(cls, n: float, k: float, freq: float, **kwargs):
```

Comment:
please add type annotations for the return `Union[Medium, Lorentz]`

Reply:
this class is a bit recursive I guess.. the `Medium` inherits from `AbstractMedium` but then `AbstractMedium.from_general_nk` returns a `Medium`. I'm not sure we even want to include this method unless it's very useful because it introduces a circular dependency and I think would break if we ever moved `Medium` and `AbstractMedium` to separate files.. is there a better way? or can we just leave out?

Reply:
Another comment, if we keep it, I feel having `from_nk_general` is better than `from_general_nk` only because then the `from_nx` methods can be next to each other alphabetically and easier to compare in IDE and in the API reference.